cpp/oneapi/dal/algo/decision_forest/common.cpp

```diff
@@ -0,0 +1,172 @@
+/*******************************************************************************
+* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/algo/decision_forest/common.hpp"
+
+namespace oneapi::dal::decision_forest {
+
+class detail::descriptor_impl : public base {
+  public:
+    double observations_per_tree_fraction;
+    double impurity_threshold;
+    double min_weight_fraction_in_leaf_node;
+    double min_impurity_decrease_in_split_node;
+
+    std::int64_t n_classes;
```

Comment:
`n_classes -> classes__count`

Reply:
replaced with class_count, to align with new name for method